### DISCUSSION TITLE
Help with understanding the different sensors

### DISCUSSION THREAD

**Discussion by badbroetchen** [wlcrs/huawei_solar#112] [General]
Hi, first of all thanks for this integration and the hard work. I'm very happy with it. For me it's working with a LAN connected inverter via Modbus TCP.

But after testing now for a while, I'm still confused about some of the senors. I've also read the wiki article about the daily solar yield.

What is the difference between the "inverter_active_power" and the "inverter_input_power"? While in the wiki it was recommended to use the "inverter_active_power" and then process it with the efficienty calculation and riemann sum, the "inverter_input_power" seems to be pretty much the same value as Huawei uses for the FusionSolar portal as "Input yield", so it seems to be good to work with that one, doesn't it? 

Later a day, both sensors have the same value, but this morning, they were quite different. Thanks for any explanation on that. And in addition, does anyone know if the efficienty losses are already considered in the Huawei portal (or using "inverter_input_power" sensor)?

**Comment by wlcrs** [comment]
Input power = Power entering the inverter on the DC-side (=solar panels)
Active power = Power exiting the inverter on the AC-side (=your home/grid)

Things get more confusing if your setup contains a battery. Power flowing to the battery will not appear in the Active Power, but power flowing from the battery is included in the input power.

The efficiency calculation + Riemann sum tries to filter out the pure yield from the solar panels, excluding the batteries, but including the inverter DC-to-AC conversion losses.

**Reply by badbroetchen** [reply]
Hi, thanks for your quick reply. So, in case of a battery setup (as mine) it would mean:
Input power = Power entering the inverter on the DC-side (=solar panels + battery discharge), and actually that makes sense, as the battery is conneced to the DC-side as well. 

So there is no possibilty to get the exact same values as in the FusionPortal for the solar yield, because of HA is demanding increasing only values for the statistics, right? 

Another idea would be to use the string values directly (Example: U=170V, I=8,55A, then P=1453,5W) plus the second string, of course. I'm going to test if that will match with the values from the portal. 

EDIT: Do you know how the FusionPortal is counting the yield over the day? In my setup the refresh interval is set to 90sec. Does it mean that Huawei also uses the Riemann sum with the 90 seconds, or does the inverter stores interenally the values with a higher resolution and provide to the portal then?

**Reply by wlcrs** [reply]
I do not know how FusionSolar computes the yield. The template "Daily Solar Yield" sensor was suggested by someone in the community because it appears to give numbers that are pretty close to those in FusionSolar.
